podcasts/DiscoverEpisodeViewModel.swift

```diff
@@ -90,15 +90,20 @@ class DiscoverEpisodeViewModel: ObservableObject {
         AnalyticsHelper.listImpression(listId: listId)
     }
 
-    public func didSelectPlayEpisode() {
+    public func didSelectPlayEpisode(completion: ((Bool) -> Void)? = nil) {
```

Comment:
I hit an issue on the main discover screen, when tapping the play button and it failing the play button isn't reset. 
I think we can just pass the button directly into this function and reset its state on failure as a simple solution:

```suggestion
    public func didSelectPlayEpisode(from button: PlayPauseLabeledButton) {
```

https://user-images.githubusercontent.com/793774/197885952-6f26fe3c-7e71-47c0-8984-b9c2bf416056.mp4